components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/config/deployer/ConfigDeployer.java

```diff
@@ -0,0 +1,252 @@
+        ApplicationConfiguration appConfig = carbonApp.getAppConfig();
+        List<Artifact.Dependency> deps = appConfig.getApplicationArtifact().getDependencies();
+
+        List<Artifact> artifacts = new ArrayList<Artifact>();
+        for (Artifact.Dependency dep : deps) {
+            if (dep.getArtifact() != null) {
+                artifacts.add(dep.getArtifact());
+            }
+        }
+        deployConfigArtifacts(artifacts, carbonApp.getAppNameWithVersion());
+    }
+
+    @Override
+    public void undeployArtifacts(CarbonApplication carbonApp, AxisConfiguration axisConfig) throws DeploymentException {
+
+    }
+
+    private void deployConfigArtifacts(List<Artifact> artifacts, String parentAppName) {
+        artifacts.stream().filter(artifact -> PROPERTY_TYPE.equals(artifact.getType())).forEach(artifact -> {
+            if (log.isDebugEnabled()) {
+                log.debug("Deploying config artifact: " + artifact.getName());
+            }
+            writePropertyToMap(artifact);
+        });
+    }
+
+    private void writePropertyToMap(Artifact artifact) {
+        // get the file path of the registry config file
+        List<CappFile> files = artifact.getFiles();
+        if (files.size() == 1) {
+            Path confFolder = Paths.get(getHome(), "conf");
+            Path globalPropertiesFilePath = confFolder.resolve(GLOBAL_CONFIG_FILE_NAME) ;
+            Path serverConfPropertyPath = confFolder.resolve(LOCAL_CONFIG_FILE_NAME);
+            String configFilePath = artifact.getExtractedPath() + File.separator + LOCAL_CONFIG_FILE_NAME;
+            processConfFile(configFilePath, globalPropertiesFilePath.toString(), serverConfPropertyPath.toString());
+        } else {
+            log.error("config/property type must have a single file which declares " +
+                    "config. But " + files.size() + " files found.");
+        }
+    }
+
+    public void processConfFile(String configFilePath, String globalPropertiesFilePath, String serverConfPropertyPath) {
+        File configFile = new File(configFilePath);
+        // Load capp conf property file
+        Properties configProperties = loadPropertiesFromFile(configFile);
+        // Load global conf property file
+        this.globalProperties = loadPropertiesFromFile(new File(globalPropertiesFilePath));
+        // Load sever conf property file
+        Properties serverConfigProperties = loadPropertiesFromFile(new File(serverConfPropertyPath));
+
+        Properties newServerConfigProperties = new Properties();
+        String isCertDeploymentEnabled = getValueOfKey(IS_CERT_DEPLOYMENT_ENABLED);
+        if (isCertDeploymentEnabled == null) {
+            isCertDeploymentEnabled = "true";
+        }
+        PropertyHolder.getInstance().setProperty(IS_CERT_DEPLOYMENT_ENABLED, isCertDeploymentEnabled);
+        if (serverConfigProperties.size() == 0 && configProperties.size() == 0 ) {
+            log.info("No configuration is used in the integration");
```

Comment:
Do we need this information as info log ? Since we are calling this method for each CAPP we might get bunch of "No configuration is used in the integration" logs.
If the log is needed, shall we append the CAPP name here